pkg/middleware/payment/middleware.go

```diff
@@ -0,0 +1,206 @@
+package payment
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log/slog"
+	"net/http"
+
+	"github.com/4chain-ag/go-bsv-middleware/pkg/internal/logging"
+	"github.com/4chain-ag/go-bsv-middleware/pkg/middleware/auth"
+	"github.com/4chain-ag/go-bsv-middleware/pkg/temporary/wallet"
+)
+
+// Middleware is the payment middleware handler that implements Direct Payment Protocol (DPP) for HTTP-based micropayments
+type Middleware struct {
+	logger                *slog.Logger
+	wallet                wallet.PaymentInterface
+	calculateRequestPrice func(r *http.Request) (int, error)
+}
+
+// New creates a new payment middleware
+func New(opts Options) (*Middleware, error) {
+	if opts.Wallet == nil {
+		return nil, ErrNoWallet
+	}
+
+	if opts.CalculateRequestPrice == nil {
+		opts.CalculateRequestPrice = DefaultPriceFunc
+	}
+
+	logger := logging.Child(nil, "payment-middleware")
```

Comment:
It would be good to allow for passing the logger in options, and make a child from it.

This way, adopter can setup the logger with his options, and you will just use it, as is.